File: frontend/src/features/actividades/realizar_entrega_ep_firmada/models/realizar_entrega_ep_firmada.ts
import type { Auditoria } from '@/models/Auditoria';

export interface RealizarEntregaEpFirmada extends Auditoria {
  id: number;
  id_expediente: number;
  id_actividad: string;
  entregado_a: string | null;
  aplica_excepcion: string | null;  // "SI" | "NO" — readonly, calculado en backend
  observaciones: string | null;
}

export interface DatosHeredadosEntregaEp {
  concepto_firma?: string | null;
  concepto_firma_descripcion?: string | null;
  notaria?: string | null;
  numero_notaria?: number | null;
  ciudad_notaria?: string | null;
  fecha_notaria?: string | null;
  numero_escritura?: string | null;
  fecha_escritura?: string | null;
  representante_legal?: string | null;
}

export interface GetByExpedienteResponse {
  formulario: RealizarEntregaEpFirmada;
  datos_heredados: DatosHeredadosEntregaEp;
}

export const EMPTY_REALIZAR_ENTREGA_EP_FIRMADA = (
  id_expediente: number,
): RealizarEntregaEpFirmada => ({
  id: 0,
  id_expediente,
  id_actividad: 'BBVA_ESCRITURACION_REALIZAR_ENTREGA_EP_FIRMADA',
  entregado_a: null,
  aplica_excepcion: null,
  observaciones: null,
  is_active: true,
  row_status: true,
  created_by: 0,
  created_date: '',
  modified_by: null,
  modified_date: null,
});

File: backend/Multibanca.Domain.Models/Multibanca/BBVA/Escrituracion/realizar_entrega_ep_firmada.cs
namespace Multibanca.Domain.Models.Multibanca.BBVA.Escrituracion;

public class realizar_entrega_ep_firmada
{
    public long     id { get; set; }
    public long     id_expediente { get; set; }
    public string?  id_actividad { get; set; }

    // Campos de la actividad
    public string?  entregado_a { get; set; }
    public string?  aplica_excepcion { get; set; }
    public string?  observaciones { get; set; }

    // Auditoría
    public bool      is_active { get; set; }
    public bool      row_status { get; set; }
    public int       created_by { get; set; }
    public DateTime  created_date { get; set; }
    public int?      modified_by { get; set; }
    public DateTime? modified_date { get; set; }
}
